File: CPP/NikhilSinghh/643. Maximum Average Subarray I.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& n, int k) {
      int a=0, b=k, sum=0, maxSum=0;
      for(int i=0; i<k; i++)
      sum+=n[i];
      maxSum=sum;
     // maxSum+=n[i];
       // sum=maxSum;
        for(int i=k; i<n.size(); i++){
                sum+=n[i];
                sum-=n[i-k];
                maxSum = max(maxSum, sum);
                    
                }
               // cout<<" SUM is "<<sum<<" ";             
        
        
        return (double)maxSum/k;
    }
};
 



// class Solution {
// public:
//     double findMaxAverage(vector<int>& n, int k) {
//       int a=0, b=k, sum=0, maxSum=n[0];
//         for(int i=0; i<=n.size()-k; i++){
//                 for(int j=i; j<i+k; j++){
//                     sum+=n[j];
//                     cout<<n[j]<<" + ";
                    
//                 }
//               //  cout<<" SUM is "<<sum<<" ";
//                 maxSum = max(maxSum, sum);
//                 sum=0;
//         }
        
//         return (double)maxSum/k;
//     }
// };

File: Java/Akash-212/643. Maximum Average Subarray I.java
class Solution {
    public double findMaxAverage(int[] arr, int k) {
      
          int n = arr.length;
        double max = 0;
        double maximum = Double.NEGATIVE_INFINITY; // Initialize to smallest possible value
        double result;
        int update = n - k;
        if (n == 1) {
            max += arr[0];
            return max;
        } else {
            for (int i = 0; i <= update; i++) {
                if (i == 0) {
                    for (int j = 0; j < k; j++) {
                        max += arr[j];
                    }
                } else {
                    max = max - arr[i - 1] + arr[i + k - 1];
                }
                result = max / k;
                maximum = Math.max(maximum, result);
            }
            return maximum;
        }
    }
}
